compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -321,7 +321,7 @@ trait QuotesAndSplices {
       ctx.error(missingArgMsg(qctx, defn.QuoteContextClass.typeRef, ""), ctx.source.atSpan(tree.span))
 
     val quoted = tree.quoted
-    val exprPt = pt.baseType(if quoted.isType then defn.QuotedTypeClass else defn.QuotedExprClass)
+    val exprPt = pt.baseType(defn.QuotedExprClass)
```

Comment:
I don't understand how this can be correct. If `pt` is `quoted.Type[...]` then `pt.baseType` is `NoType` and therefore `quotedPt` is `Any`. Is that what we want? If yes, it would be good to express that logic more directly instead of relying on a `baseType` to fail.

Reply:
Added it again